crypto/src/error.rs

```diff
@@ -0,0 +1,11 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum MerkleTreeError {
+    #[error("Merkle tree too large")]
+    TooLarge,
+    #[error("Already at root")]
+    RootHasNoParent,
+    #[error("Already at leaf")]
+    LeafHasNoChild,
+}
```

Comment:
I wonder whether we should try to keep all of the errors in one place (here, `crate::error::MerkleTreeError`), or to keep the errors near the functionality they relate to (in this case, `crate::merkle::Error`). I kind of lean towards the latter, since it means that the errors are together with the code that creates them, and the names can be factored into the module path (as in the stdlib, where you have `std::io::Error`, etc).